src/api/app/controllers/status_project_controller.rb

```diff
@@ -0,0 +1,42 @@
+class StatusProjectController < ApplicationController
+  def show
+    dbproj = Project.get_by_name(params[:project])
+    @packages = ProjectStatus::Calculator.new(dbproj).calc_status
+    find_relationships_for_packages(@packages)
+  end
+
+  private
+
+  def role_from_cache(role_id)
+    @rolecache[role_id] || (@rolecache[role_id] = Role.find(role_id).title)
+  end
+
+  def user_from_cache(user_id)
+    @usercache[user_id] || (@usercache[user_id] = User.find(user_id).login)
+  end
+
+  def group_from_cache(group_id)
+    @groupcache[group_id] || (@groupcache[group_id] = Group.find(group_id).title)
+  end
+
+  def find_relationships_for_packages(packages)
+    package_hash = {}
+    packages.each_value do |p|
+      package_hash[p.package_id] = p
+      package_hash[p.develpack.package_id] = p.develpack if p.develpack
+    end
+    @rolecache = {}
```

Comment:
I would move the declaration of these instance variables in their respective `*_from_cache` methods. It seems a bit out of place here.

Reply:
this review should have been done here https://github.com/openSUSE/open-build-service/pull/5765/files :sob: :sob: :sob:

Reply:
The PR you linked didn't touch the `*_from_cache` methods.

Reply:
@dmarcoux it was moved around, like I'm doing now :satisfied: